### ISSUE TITLE
Make notification targeting more intelligent.

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-meta#15]
Right now notifications go to every device, so regardless of whether you were actively engaged in the conversation on one device, every other device is going to have a stack of notifications waiting for you (worse - if you're typing on Riot Web your phone will be buzzing away in the background like a crazy thing).

The scope of this project is to:
 - define what the optimal behaviour is
 - implement across all Riot clients

**Comment by ejpcmac** [comment]
It would be great to be able to tune more precisely mobile notifications. For instance, say I want to receive notifications for all messages in a room on desktop, but only for mentions on mobile.

This setting would be per-room. In the desktop app, something like a second cursor in the room notifications parameters could enable this.

**Comment by tgr** [comment]
Yeah, this is annoying in general, and super disruptive in an office setting (even buzzing would be bad but riot-android defaults to an audible ping, not just buzzing). A private chat sounds like an alarm clock going off in your pocket.

Maybe the mobile apps could delay notifications a few seconds after the arrival of the message, so there is a chance for read receipts from other devices to arrive first?
